In [1]:
from scipy.io import loadmat
import pathlib
import shutil
import os

### 1. Create Folders

In [2]:
label = loadmat('raw/devkit/cars_meta.mat')
class_names = label['class_names'][0]
for class_name in class_names:
    print('--', class_name[0], '--')
    class_name[0] = class_name[0].replace("/","_")
    path_train = 'data/train/'+class_name[0]
    path_test = 'data/test/'+class_name[0]
    path_val = 'data/valid/'+class_name[0]
    pathlib.Path(path_train).mkdir(parents=True, exist_ok=True)
    pathlib.Path(path_test).mkdir(parents=True, exist_ok=True)
    pathlib.Path(path_val).mkdir(parents=True, exist_ok=True)

-- AM General Hummer SUV 2000 --
-- Acura RL Sedan 2012 --
-- Acura TL Sedan 2012 --
-- Acura TL Type-S 2008 --
-- Acura TSX Sedan 2012 --
-- Acura Integra Type R 2001 --
-- Acura ZDX Hatchback 2012 --
-- Aston Martin V8 Vantage Convertible 2012 --
-- Aston Martin V8 Vantage Coupe 2012 --
-- Aston Martin Virage Convertible 2012 --
-- Aston Martin Virage Coupe 2012 --
-- Audi RS 4 Convertible 2008 --
-- Audi A5 Coupe 2012 --
-- Audi TTS Coupe 2012 --
-- Audi R8 Coupe 2012 --
-- Audi V8 Sedan 1994 --
-- Audi 100 Sedan 1994 --
-- Audi 100 Wagon 1994 --
-- Audi TT Hatchback 2011 --
-- Audi S6 Sedan 2011 --
-- Audi S5 Convertible 2012 --
-- Audi S5 Coupe 2012 --
-- Audi S4 Sedan 2012 --
-- Audi S4 Sedan 2007 --
-- Audi TT RS Coupe 2012 --
-- BMW ActiveHybrid 5 Sedan 2012 --
-- BMW 1 Series Convertible 2012 --
-- BMW 1 Series Coupe 2012 --
-- BMW 3 Series Sedan 2012 --
-- BMW 3 Series Wagon 2012 --
-- BMW 6 Series Convertible 2007 --
-- BMW X5 SUV 2007 --
-- BMW X6 SUV 2012 --
-- BMW M3 Coup

### 2. Copy Images to Folders

In [3]:
def copy_images(mat_path, raw_path, new_path):
    train_label = loadmat(mat_path)
    classes = train_label['annotations'][0]
    for i in range(len(classes)):
        class_name_index = classes[i]['class'][0][0] - 1
        folder_name = class_names[class_name_index][0]
        file_name = classes[i]['fname'][0]
        shutil.copyfile(raw_path+file_name, new_path+folder_name+'/'+file_name)
        print((i+1), class_names[class_name_index][0])

copy_images('raw/devkit/cars_train_annos.mat', 'raw/cars_train/', 'data/train/')
copy_images('raw/cars_test_annos_withlabels.mat', 'raw/cars_test/', 'data/valid/')


1 Audi TTS Coupe 2012
2 Acura TL Sedan 2012
3 Dodge Dakota Club Cab 2007
4 Hyundai Sonata Hybrid Sedan 2012
5 Ford F-450 Super Duty Crew Cab 2012
6 Geo Metro Convertible 1993
7 Dodge Journey SUV 2012
8 Dodge Charger Sedan 2012
9 Mitsubishi Lancer Sedan 2012
10 Chevrolet Traverse SUV 2012
11 Buick Verano Sedan 2012
12 Toyota Sequoia SUV 2012
13 Hyundai Elantra Sedan 2007
14 Dodge Caravan Minivan 1997
15 Volvo C30 Hatchback 2012
16 Plymouth Neon Coupe 1999
17 Audi TTS Coupe 2012
18 Chevrolet Malibu Sedan 2007
19 Volkswagen Beetle Hatchback 2012
20 Chevrolet Corvette Ron Fellows Edition Z06 2007
21 Chrysler 300 SRT-8 2010
22 BMW M6 Convertible 2010
23 GMC Yukon Hybrid SUV 2012
24 Nissan Juke Hatchback 2012
25 Volvo 240 Sedan 1993
26 Hyundai Sonata Hybrid Sedan 2012
27 Suzuki SX4 Sedan 2012
28 Dodge Ram Pickup 3500 Crew Cab 2010
29 Spyker C8 Coupe 2009
30 Volvo 240 Sedan 1993
31 Land Rover Range Rover SUV 2012
32 Hyundai Elantra Touring Hatchback 2012
33 Chevrolet Cobalt SS 2010
34 Hyundai

406 Chevrolet Silverado 1500 Regular Cab 2012
407 Chevrolet Camaro Convertible 2012
408 Audi TT Hatchback 2011
409 Toyota 4Runner SUV 2012
410 Acura TSX Sedan 2012
411 Audi TTS Coupe 2012
412 Rolls-Royce Ghost Sedan 2012
413 Chrysler Crossfire Convertible 2008
414 Dodge Ram Pickup 3500 Crew Cab 2010
415 Chevrolet Corvette Convertible 2012
416 Ford GT Coupe 2006
417 Buick Enclave SUV 2012
418 Chevrolet Corvette Ron Fellows Edition Z06 2007
419 Chevrolet Malibu Sedan 2007
420 FIAT 500 Abarth 2012
421 Lamborghini Gallardo LP 570-4 Superleggera 2012
422 Mercedes-Benz C-Class Sedan 2012
423 Dodge Caravan Minivan 1997
424 Acura TSX Sedan 2012
425 Acura ZDX Hatchback 2012
426 Volvo 240 Sedan 1993
427 Ford Fiesta Sedan 2012
428 Spyker C8 Convertible 2009
429 Chevrolet Malibu Sedan 2007
430 Tesla Model S Sedan 2012
431 Toyota Camry Sedan 2012
432 Mercedes-Benz SL-Class Coupe 2009
433 Mercedes-Benz E-Class Sedan 2012
434 Nissan 240SX Coupe 1998
435 Chevrolet Express Cargo Van 2007
436 Cadillac C

843 Mazda Tribute SUV 2011
844 Nissan 240SX Coupe 1998
845 Toyota Corolla Sedan 2012
846 Dodge Ram Pickup 3500 Crew Cab 2010
847 Mazda Tribute SUV 2011
848 McLaren MP4-12C Coupe 2012
849 Chevrolet HHR SS 2010
850 Hyundai Sonata Hybrid Sedan 2012
851 Volvo XC90 SUV 2007
852 Volvo XC90 SUV 2007
853 Lamborghini Diablo Coupe 2001
854 Chevrolet Express Van 2007
855 Hyundai Genesis Sedan 2012
856 Bentley Mulsanne Sedan 2011
857 GMC Terrain SUV 2012
858 Acura RL Sedan 2012
859 BMW 3 Series Wagon 2012
860 Ferrari 458 Italia Coupe 2012
861 Ferrari California Convertible 2012
862 GMC Savana Van 2012
863 Chrysler Sebring Convertible 2010
864 Ferrari 458 Italia Convertible 2012
865 Audi 100 Sedan 1994
866 Cadillac SRX SUV 2012
867 Dodge Ram Pickup 3500 Quad Cab 2009
868 Dodge Sprinter Cargo Van 2009
869 Toyota Camry Sedan 2012
870 BMW 1 Series Coupe 2012
871 Bugatti Veyron 16.4 Convertible 2009
872 Chevrolet Corvette Convertible 2012
873 Mercedes-Benz SL-Class Coupe 2009
874 Chevrolet Silverado 15

1248 McLaren MP4-12C Coupe 2012
1249 Bugatti Veyron 16.4 Coupe 2009
1250 Suzuki Kizashi Sedan 2012
1251 Audi V8 Sedan 1994
1252 smart fortwo Convertible 2012
1253 BMW X3 SUV 2012
1254 Land Rover LR2 SUV 2012
1255 Acura Integra Type R 2001
1256 Chevrolet Express Van 2007
1257 Hyundai Veracruz SUV 2012
1258 Dodge Durango SUV 2007
1259 Dodge Journey SUV 2012
1260 Dodge Durango SUV 2007
1261 Dodge Caravan Minivan 1997
1262 Dodge Caliber Wagon 2007
1263 Isuzu Ascender SUV 2008
1264 Acura TL Type-S 2008
1265 Tesla Model S Sedan 2012
1266 GMC Canyon Extended Cab 2012
1267 Honda Odyssey Minivan 2007
1268 Chevrolet Silverado 1500 Hybrid Crew Cab 2012
1269 Spyker C8 Coupe 2009
1270 Bentley Continental Flying Spur Sedan 2007
1271 Maybach Landaulet Convertible 2012
1272 Hyundai Tucson SUV 2012
1273 Ferrari California Convertible 2012
1274 Chevrolet Monte Carlo Coupe 2007
1275 BMW 3 Series Sedan 2012
1276 Aston Martin V8 Vantage Coupe 2012
1277 AM General Hummer SUV 2000
1278 Chrysler Crossfire Con

1659 Hyundai Veloster Hatchback 2012
1660 FIAT 500 Convertible 2012
1661 AM General Hummer SUV 2000
1662 Porsche Panamera Sedan 2012
1663 BMW 1 Series Convertible 2012
1664 Honda Odyssey Minivan 2012
1665 Hyundai Sonata Sedan 2012
1666 Lamborghini Gallardo LP 570-4 Superleggera 2012
1667 Jeep Wrangler SUV 2012
1668 GMC Savana Van 2012
1669 Hyundai Veracruz SUV 2012
1670 Volkswagen Golf Hatchback 2012
1671 Buick Regal GS 2012
1672 BMW X6 SUV 2012
1673 Chrysler PT Cruiser Convertible 2008
1674 Mitsubishi Lancer Sedan 2012
1675 BMW M5 Sedan 2010
1676 Eagle Talon Hatchback 1998
1677 Bentley Continental Flying Spur Sedan 2007
1678 Aston Martin Virage Convertible 2012
1679 Chevrolet Express Cargo Van 2007
1680 Aston Martin V8 Vantage Coupe 2012
1681 BMW X6 SUV 2012
1682 Dodge Caliber Wagon 2012
1683 Dodge Ram Pickup 3500 Crew Cab 2010
1684 Lamborghini Aventador Coupe 2012
1685 Dodge Challenger SRT8 2011
1686 Ford F-150 Regular Cab 2012
1687 BMW 3 Series Wagon 2012
1688 Ford F-450 Super Duty 

2073 McLaren MP4-12C Coupe 2012
2074 Dodge Ram Pickup 3500 Crew Cab 2010
2075 Chrysler Sebring Convertible 2010
2076 Mercedes-Benz 300-Class Convertible 1993
2077 Chrysler Sebring Convertible 2010
2078 Dodge Charger Sedan 2012
2079 Chevrolet Silverado 1500 Regular Cab 2012
2080 Hyundai Veracruz SUV 2012
2081 BMW X5 SUV 2007
2082 Chevrolet Traverse SUV 2012
2083 Lamborghini Gallardo LP 570-4 Superleggera 2012
2084 Lamborghini Aventador Coupe 2012
2085 Ford E-Series Wagon Van 2012
2086 Volkswagen Beetle Hatchback 2012
2087 BMW 3 Series Wagon 2012
2088 GMC Terrain SUV 2012
2089 Geo Metro Convertible 1993
2090 Land Rover Range Rover SUV 2012
2091 Chevrolet Monte Carlo Coupe 2007
2092 Suzuki SX4 Sedan 2012
2093 Ford F-150 Regular Cab 2012
2094 Jeep Wrangler SUV 2012
2095 Acura Integra Type R 2001
2096 Bentley Continental Flying Spur Sedan 2007
2097 Chevrolet Silverado 1500 Regular Cab 2012
2098 Chevrolet Traverse SUV 2012
2099 Chevrolet Avalanche Crew Cab 2012
2100 Dodge Journey SUV 2012
21

2479 Dodge Caliber Wagon 2012
2480 BMW M3 Coupe 2012
2481 Suzuki Kizashi Sedan 2012
2482 HUMMER H2 SUT Crew Cab 2009
2483 Dodge Sprinter Cargo Van 2009
2484 Aston Martin Virage Convertible 2012
2485 Ford Expedition EL SUV 2009
2486 Chevrolet TrailBlazer SS 2009
2487 Jaguar XK XKR 2012
2488 Dodge Ram Pickup 3500 Quad Cab 2009
2489 Lamborghini Diablo Coupe 2001
2490 Ford Expedition EL SUV 2009
2491 Acura TL Sedan 2012
2492 Hyundai Elantra Sedan 2007
2493 Bentley Continental Flying Spur Sedan 2007
2494 Dodge Ram Pickup 3500 Crew Cab 2010
2495 Tesla Model S Sedan 2012
2496 GMC Terrain SUV 2012
2497 Jaguar XK XKR 2012
2498 Ford Freestar Minivan 2007
2499 GMC Terrain SUV 2012
2500 Hyundai Elantra Sedan 2007
2501 Maybach Landaulet Convertible 2012
2502 Chevrolet Malibu Hybrid Sedan 2010
2503 Chevrolet TrailBlazer SS 2009
2504 BMW 1 Series Coupe 2012
2505 Audi 100 Wagon 1994
2506 Ford Freestar Minivan 2007
2507 Mercedes-Benz S-Class Sedan 2012
2508 Ford Expedition EL SUV 2009
2509 BMW X6 SUV 2

2892 Bentley Continental GT Coupe 2007
2893 Dodge Ram Pickup 3500 Crew Cab 2010
2894 Infiniti G Coupe IPL 2012
2895 Hyundai Veloster Hatchback 2012
2896 Audi TTS Coupe 2012
2897 Acura TSX Sedan 2012
2898 Chrysler Sebring Convertible 2010
2899 Audi A5 Coupe 2012
2900 Audi V8 Sedan 1994
2901 BMW X3 SUV 2012
2902 MINI Cooper Roadster Convertible 2012
2903 Chevrolet Silverado 1500 Regular Cab 2012
2904 Spyker C8 Coupe 2009
2905 Chevrolet Sonic Sedan 2012
2906 Buick Verano Sedan 2012
2907 Jaguar XK XKR 2012
2908 Honda Accord Coupe 2012
2909 Chevrolet Impala Sedan 2007
2910 Chevrolet Corvette ZR1 2012
2911 Acura Integra Type R 2001
2912 Chevrolet Silverado 1500 Extended Cab 2012
2913 Acura TL Type-S 2008
2914 Mercedes-Benz C-Class Sedan 2012
2915 Spyker C8 Coupe 2009
2916 Ram C_V Cargo Van Minivan 2012
2917 Eagle Talon Hatchback 1998
2918 BMW M6 Convertible 2010
2919 Acura Integra Type R 2001
2920 HUMMER H3T Crew Cab 2010
2921 Aston Martin Virage Coupe 2012
2922 Spyker C8 Convertible 2009
29

3279 Hyundai Tucson SUV 2012
3280 Dodge Caliber Wagon 2007
3281 Toyota 4Runner SUV 2012
3282 Chrysler 300 SRT-8 2010
3283 Audi R8 Coupe 2012
3284 Suzuki SX4 Hatchback 2012
3285 Bentley Mulsanne Sedan 2011
3286 Mercedes-Benz Sprinter Van 2012
3287 Buick Rainier SUV 2007
3288 BMW M5 Sedan 2010
3289 Audi TTS Coupe 2012
3290 Hyundai Genesis Sedan 2012
3291 Chevrolet Corvette ZR1 2012
3292 Hyundai Tucson SUV 2012
3293 FIAT 500 Convertible 2012
3294 BMW M3 Coupe 2012
3295 Dodge Dakota Club Cab 2007
3296 Nissan 240SX Coupe 1998
3297 McLaren MP4-12C Coupe 2012
3298 BMW M5 Sedan 2010
3299 Bentley Mulsanne Sedan 2011
3300 Volkswagen Golf Hatchback 1991
3301 Nissan 240SX Coupe 1998
3302 Mercedes-Benz S-Class Sedan 2012
3303 Land Rover LR2 SUV 2012
3304 Chevrolet Sonic Sedan 2012
3305 Dodge Ram Pickup 3500 Quad Cab 2009
3306 Ford Edge SUV 2012
3307 Volvo XC90 SUV 2007
3308 Acura ZDX Hatchback 2012
3309 GMC Canyon Extended Cab 2012
3310 Spyker C8 Convertible 2009
3311 Aston Martin V8 Vantage Conver

3704 Jeep Liberty SUV 2012
3705 Hyundai Elantra Sedan 2007
3706 HUMMER H3T Crew Cab 2010
3707 FIAT 500 Convertible 2012
3708 Cadillac Escalade EXT Crew Cab 2007
3709 BMW 3 Series Wagon 2012
3710 Hyundai Elantra Sedan 2007
3711 Plymouth Neon Coupe 1999
3712 Dodge Caliber Wagon 2007
3713 MINI Cooper Roadster Convertible 2012
3714 Fisker Karma Sedan 2012
3715 Acura Integra Type R 2001
3716 BMW 6 Series Convertible 2007
3717 Audi 100 Sedan 1994
3718 Buick Rainier SUV 2007
3719 Porsche Panamera Sedan 2012
3720 Rolls-Royce Phantom Drophead Coupe Convertible 2012
3721 Audi S5 Coupe 2012
3722 BMW 3 Series Sedan 2012
3723 Audi RS 4 Convertible 2008
3724 Chevrolet TrailBlazer SS 2009
3725 Ford F-150 Regular Cab 2007
3726 Ferrari California Convertible 2012
3727 Jeep Liberty SUV 2012
3728 smart fortwo Convertible 2012
3729 Honda Odyssey Minivan 2007
3730 Dodge Dakota Club Cab 2007
3731 Isuzu Ascender SUV 2008
3732 Aston Martin V8 Vantage Coupe 2012
3733 Chevrolet Corvette Ron Fellows Edition Z06 

4116 Chrysler Sebring Convertible 2010
4117 Cadillac SRX SUV 2012
4118 Dodge Caliber Wagon 2012
4119 Suzuki Kizashi Sedan 2012
4120 Bugatti Veyron 16.4 Coupe 2009
4121 Aston Martin V8 Vantage Convertible 2012
4122 Ford E-Series Wagon Van 2012
4123 Ford Focus Sedan 2007
4124 Chevrolet Malibu Hybrid Sedan 2010
4125 Mazda Tribute SUV 2011
4126 Bentley Continental GT Coupe 2012
4127 Dodge Caliber Wagon 2012
4128 Volkswagen Beetle Hatchback 2012
4129 BMW 3 Series Sedan 2012
4130 Maybach Landaulet Convertible 2012
4131 Hyundai Veloster Hatchback 2012
4132 Hyundai Elantra Sedan 2007
4133 Honda Accord Coupe 2012
4134 Jeep Liberty SUV 2012
4135 Mitsubishi Lancer Sedan 2012
4136 Mercedes-Benz E-Class Sedan 2012
4137 Chevrolet Traverse SUV 2012
4138 BMW 6 Series Convertible 2007
4139 Geo Metro Convertible 1993
4140 Ford F-150 Regular Cab 2012
4141 Suzuki SX4 Sedan 2012
4142 GMC Terrain SUV 2012
4143 Nissan 240SX Coupe 1998
4144 Ford Edge SUV 2012
4145 Honda Accord Coupe 2012
4146 Suzuki Kizashi S

4532 Infiniti QX56 SUV 2011
4533 Audi RS 4 Convertible 2008
4534 Volvo XC90 SUV 2007
4535 Hyundai Elantra Sedan 2007
4536 Ford Fiesta Sedan 2012
4537 Audi TTS Coupe 2012
4538 Volvo 240 Sedan 1993
4539 Chevrolet Monte Carlo Coupe 2007
4540 Jaguar XK XKR 2012
4541 Dodge Charger Sedan 2012
4542 Nissan 240SX Coupe 1998
4543 Dodge Charger Sedan 2012
4544 AM General Hummer SUV 2000
4545 Ferrari California Convertible 2012
4546 Volkswagen Beetle Hatchback 2012
4547 BMW 3 Series Wagon 2012
4548 BMW 1 Series Convertible 2012
4549 Audi 100 Sedan 1994
4550 Cadillac CTS-V Sedan 2012
4551 BMW 1 Series Coupe 2012
4552 Aston Martin Virage Convertible 2012
4553 Buick Rainier SUV 2007
4554 Land Rover LR2 SUV 2012
4555 Toyota Camry Sedan 2012
4556 Buick Rainier SUV 2007
4557 Infiniti QX56 SUV 2011
4558 GMC Savana Van 2012
4559 GMC Savana Van 2012
4560 Chevrolet Camaro Convertible 2012
4561 Honda Accord Sedan 2012
4562 Spyker C8 Coupe 2009
4563 HUMMER H2 SUT Crew Cab 2009
4564 GMC Terrain SUV 2012
4565 C

4925 Ford Ranger SuperCab 2011
4926 Jaguar XK XKR 2012
4927 BMW Z4 Convertible 2012
4928 Ford F-150 Regular Cab 2012
4929 Bentley Continental GT Coupe 2012
4930 Hyundai Genesis Sedan 2012
4931 Audi V8 Sedan 1994
4932 Hyundai Azera Sedan 2012
4933 Suzuki Aerio Sedan 2007
4934 Nissan NV Passenger Van 2012
4935 MINI Cooper Roadster Convertible 2012
4936 Chevrolet Monte Carlo Coupe 2007
4937 Hyundai Elantra Touring Hatchback 2012
4938 Dodge Ram Pickup 3500 Crew Cab 2010
4939 Fisker Karma Sedan 2012
4940 Audi S4 Sedan 2012
4941 Mercedes-Benz 300-Class Convertible 1993
4942 Aston Martin V8 Vantage Convertible 2012
4943 Chevrolet Monte Carlo Coupe 2007
4944 Audi TT Hatchback 2011
4945 BMW M3 Coupe 2012
4946 BMW 1 Series Convertible 2012
4947 Ford Edge SUV 2012
4948 Dodge Charger Sedan 2012
4949 Audi 100 Sedan 1994
4950 Dodge Ram Pickup 3500 Quad Cab 2009
4951 Bentley Continental Flying Spur Sedan 2007
4952 Lamborghini Gallardo LP 570-4 Superleggera 2012
4953 Eagle Talon Hatchback 1998
4954 Ch

5349 Acura ZDX Hatchback 2012
5350 Ram C_V Cargo Van Minivan 2012
5351 Bugatti Veyron 16.4 Convertible 2009
5352 Hyundai Tucson SUV 2012
5353 Audi A5 Coupe 2012
5354 Plymouth Neon Coupe 1999
5355 Nissan NV Passenger Van 2012
5356 Hyundai Elantra Touring Hatchback 2012
5357 GMC Savana Van 2012
5358 GMC Yukon Hybrid SUV 2012
5359 Audi TT Hatchback 2011
5360 Ferrari 458 Italia Convertible 2012
5361 Toyota Corolla Sedan 2012
5362 Ford Fiesta Sedan 2012
5363 Ford E-Series Wagon Van 2012
5364 Plymouth Neon Coupe 1999
5365 Land Rover Range Rover SUV 2012
5366 Aston Martin Virage Convertible 2012
5367 Volvo C30 Hatchback 2012
5368 Chevrolet Avalanche Crew Cab 2012
5369 Ford F-150 Regular Cab 2012
5370 Chevrolet Silverado 1500 Hybrid Crew Cab 2012
5371 Mercedes-Benz C-Class Sedan 2012
5372 Honda Odyssey Minivan 2007
5373 Aston Martin V8 Vantage Convertible 2012
5374 Buick Regal GS 2012
5375 GMC Yukon Hybrid SUV 2012
5376 Volvo 240 Sedan 1993
5377 Ford Focus Sedan 2007
5378 Chevrolet Malibu Seda

5754 AM General Hummer SUV 2000
5755 Hyundai Azera Sedan 2012
5756 Lincoln Town Car Sedan 2011
5757 HUMMER H3T Crew Cab 2010
5758 Buick Enclave SUV 2012
5759 Chevrolet Sonic Sedan 2012
5760 Bugatti Veyron 16.4 Convertible 2009
5761 Porsche Panamera Sedan 2012
5762 Chrysler PT Cruiser Convertible 2008
5763 Bentley Mulsanne Sedan 2011
5764 Chrysler Town and Country Minivan 2012
5765 Dodge Charger SRT-8 2009
5766 Volkswagen Golf Hatchback 2012
5767 Chrysler Sebring Convertible 2010
5768 Dodge Charger SRT-8 2009
5769 Bentley Continental GT Coupe 2012
5770 Scion xD Hatchback 2012
5771 GMC Terrain SUV 2012
5772 BMW X3 SUV 2012
5773 Hyundai Veloster Hatchback 2012
5774 Rolls-Royce Ghost Sedan 2012
5775 Geo Metro Convertible 1993
5776 Jeep Patriot SUV 2012
5777 Toyota Corolla Sedan 2012
5778 Mazda Tribute SUV 2011
5779 Bentley Continental Flying Spur Sedan 2007
5780 Jeep Compass SUV 2012
5781 Chevrolet Malibu Sedan 2007
5782 FIAT 500 Convertible 2012
5783 Ford Edge SUV 2012
5784 Dodge Journey 

6168 Tesla Model S Sedan 2012
6169 Audi TT RS Coupe 2012
6170 Suzuki SX4 Hatchback 2012
6171 Land Rover Range Rover SUV 2012
6172 Dodge Caliber Wagon 2012
6173 Chevrolet Avalanche Crew Cab 2012
6174 AM General Hummer SUV 2000
6175 Suzuki Aerio Sedan 2007
6176 Chevrolet Silverado 1500 Regular Cab 2012
6177 Jeep Patriot SUV 2012
6178 Chevrolet Malibu Sedan 2007
6179 Acura Integra Type R 2001
6180 Chevrolet Silverado 1500 Extended Cab 2012
6181 Bentley Mulsanne Sedan 2011
6182 Fisker Karma Sedan 2012
6183 Acura TSX Sedan 2012
6184 Ford Focus Sedan 2007
6185 Jeep Patriot SUV 2012
6186 Lamborghini Reventon Coupe 2008
6187 Dodge Journey SUV 2012
6188 Volkswagen Beetle Hatchback 2012
6189 Ford Expedition EL SUV 2009
6190 Dodge Charger Sedan 2012
6191 Buick Rainier SUV 2007
6192 Hyundai Veracruz SUV 2012
6193 Audi S5 Coupe 2012
6194 Hyundai Veracruz SUV 2012
6195 Jeep Wrangler SUV 2012
6196 Acura Integra Type R 2001
6197 Buick Enclave SUV 2012
6198 MINI Cooper Roadster Convertible 2012
6199 Ca

6591 Ford F-450 Super Duty Crew Cab 2012
6592 Chevrolet Silverado 1500 Classic Extended Cab 2007
6593 Audi TT RS Coupe 2012
6594 Fisker Karma Sedan 2012
6595 Dodge Caliber Wagon 2012
6596 Dodge Caliber Wagon 2012
6597 Ford Freestar Minivan 2007
6598 BMW X6 SUV 2012
6599 Dodge Magnum Wagon 2008
6600 Audi TTS Coupe 2012
6601 HUMMER H2 SUT Crew Cab 2009
6602 Suzuki SX4 Hatchback 2012
6603 Acura ZDX Hatchback 2012
6604 Nissan NV Passenger Van 2012
6605 BMW X5 SUV 2007
6606 Lamborghini Gallardo LP 570-4 Superleggera 2012
6607 Honda Odyssey Minivan 2007
6608 smart fortwo Convertible 2012
6609 Ford Ranger SuperCab 2011
6610 Audi 100 Wagon 1994
6611 Chevrolet Silverado 1500 Hybrid Crew Cab 2012
6612 Chevrolet Monte Carlo Coupe 2007
6613 Ford GT Coupe 2006
6614 Acura TL Type-S 2008
6615 Ford GT Coupe 2006
6616 Audi S4 Sedan 2012
6617 Jaguar XK XKR 2012
6618 Chevrolet Impala Sedan 2007
6619 Ford Focus Sedan 2007
6620 BMW X5 SUV 2007
6621 Jeep Patriot SUV 2012
6622 Rolls-Royce Phantom Drophead Co

7020 Dodge Journey SUV 2012
7021 Dodge Durango SUV 2007
7022 BMW 1 Series Coupe 2012
7023 Ford E-Series Wagon Van 2012
7024 McLaren MP4-12C Coupe 2012
7025 AM General Hummer SUV 2000
7026 smart fortwo Convertible 2012
7027 BMW Z4 Convertible 2012
7028 Audi 100 Wagon 1994
7029 Cadillac CTS-V Sedan 2012
7030 Dodge Charger SRT-8 2009
7031 BMW 1 Series Coupe 2012
7032 Ford Edge SUV 2012
7033 Hyundai Elantra Touring Hatchback 2012
7034 BMW M3 Coupe 2012
7035 Chevrolet Silverado 1500 Regular Cab 2012
7036 Chevrolet Monte Carlo Coupe 2007
7037 Suzuki Aerio Sedan 2007
7038 Jeep Liberty SUV 2012
7039 Audi S5 Coupe 2012
7040 BMW 6 Series Convertible 2007
7041 Ford Fiesta Sedan 2012
7042 Rolls-Royce Phantom Drophead Coupe Convertible 2012
7043 Ferrari California Convertible 2012
7044 Dodge Dakota Crew Cab 2010
7045 BMW 3 Series Wagon 2012
7046 Chrysler Crossfire Convertible 2008
7047 Chevrolet Express Van 2007
7048 Aston Martin Virage Coupe 2012
7049 Volkswagen Golf Hatchback 1991
7050 Acura TL T

7421 Nissan Leaf Hatchback 2012
7422 GMC Yukon Hybrid SUV 2012
7423 Audi TTS Coupe 2012
7424 Toyota Sequoia SUV 2012
7425 Chevrolet Corvette Ron Fellows Edition Z06 2007
7426 Cadillac Escalade EXT Crew Cab 2007
7427 Ford Fiesta Sedan 2012
7428 Hyundai Santa Fe SUV 2012
7429 Chrysler PT Cruiser Convertible 2008
7430 Dodge Durango SUV 2012
7431 Chevrolet Corvette Convertible 2012
7432 Chevrolet Malibu Hybrid Sedan 2010
7433 Bentley Mulsanne Sedan 2011
7434 Hyundai Veracruz SUV 2012
7435 Tesla Model S Sedan 2012
7436 Hyundai Santa Fe SUV 2012
7437 Chevrolet Cobalt SS 2010
7438 Plymouth Neon Coupe 1999
7439 Audi RS 4 Convertible 2008
7440 Cadillac CTS-V Sedan 2012
7441 Volvo C30 Hatchback 2012
7442 Dodge Ram Pickup 3500 Crew Cab 2010
7443 Honda Accord Sedan 2012
7444 Nissan 240SX Coupe 1998
7445 Acura TL Type-S 2008
7446 Ford Fiesta Sedan 2012
7447 Dodge Durango SUV 2007
7448 Plymouth Neon Coupe 1999
7449 Lamborghini Reventon Coupe 2008
7450 Acura TSX Sedan 2012
7451 Bugatti Veyron 16.4 Co

7853 Cadillac SRX SUV 2012
7854 Audi S5 Convertible 2012
7855 Dodge Durango SUV 2007
7856 Lincoln Town Car Sedan 2011
7857 Mercedes-Benz Sprinter Van 2012
7858 Volvo 240 Sedan 1993
7859 Plymouth Neon Coupe 1999
7860 Acura Integra Type R 2001
7861 Chevrolet Tahoe Hybrid SUV 2012
7862 Geo Metro Convertible 1993
7863 Chevrolet Silverado 1500 Extended Cab 2012
7864 Hyundai Tucson SUV 2012
7865 Lamborghini Diablo Coupe 2001
7866 Volkswagen Golf Hatchback 1991
7867 Bugatti Veyron 16.4 Convertible 2009
7868 Chevrolet Silverado 2500HD Regular Cab 2012
7869 Dodge Magnum Wagon 2008
7870 Ferrari 458 Italia Coupe 2012
7871 Chevrolet Silverado 1500 Regular Cab 2012
7872 Jeep Grand Cherokee SUV 2012
7873 Jeep Patriot SUV 2012
7874 Buick Rainier SUV 2007
7875 Ford Ranger SuperCab 2011
7876 Chevrolet Corvette ZR1 2012
7877 McLaren MP4-12C Coupe 2012
7878 Audi S4 Sedan 2007
7879 Audi TT Hatchback 2011
7880 BMW ActiveHybrid 5 Sedan 2012
7881 Hyundai Veloster Hatchback 2012
7882 McLaren MP4-12C Coupe 201

1 Suzuki Aerio Sedan 2007
2 Ferrari 458 Italia Convertible 2012
3 Jeep Patriot SUV 2012
4 Toyota Camry Sedan 2012
5 Tesla Model S Sedan 2012
6 Chrysler Town and Country Minivan 2012
7 GMC Terrain SUV 2012
8 Mercedes-Benz S-Class Sedan 2012
9 BMW X5 SUV 2007
10 Chevrolet HHR SS 2010
11 Buick Verano Sedan 2012
12 Ford Freestar Minivan 2007
13 Ford E-Series Wagon Van 2012
14 Hyundai Elantra Sedan 2007
15 Dodge Caliber Wagon 2012
16 Cadillac CTS-V Sedan 2012
17 Land Rover Range Rover SUV 2012
18 BMW X6 SUV 2012
19 Audi S5 Coupe 2012
20 BMW X5 SUV 2007
21 Lamborghini Aventador Coupe 2012
22 Acura TL Type-S 2008
23 Chrysler PT Cruiser Convertible 2008
24 Ford Mustang Convertible 2007
25 Mercedes-Benz S-Class Sedan 2012
26 Volkswagen Golf Hatchback 1991
27 Nissan Juke Hatchback 2012
28 Volvo XC90 SUV 2007
29 Dodge Dakota Club Cab 2007
30 Acura TSX Sedan 2012
31 Mercedes-Benz E-Class Sedan 2012
32 Jeep Grand Cherokee SUV 2012
33 Toyota Corolla Sedan 2012
34 Bentley Continental Flying Spur Seda

425 Bentley Continental GT Coupe 2007
426 Aston Martin V8 Vantage Convertible 2012
427 Chevrolet Tahoe Hybrid SUV 2012
428 Mercedes-Benz S-Class Sedan 2012
429 Jeep Grand Cherokee SUV 2012
430 Mercedes-Benz S-Class Sedan 2012
431 Rolls-Royce Phantom Sedan 2012
432 Dodge Journey SUV 2012
433 Audi S6 Sedan 2011
434 BMW 3 Series Sedan 2012
435 Volvo XC90 SUV 2007
436 Jeep Wrangler SUV 2012
437 Audi S4 Sedan 2012
438 Ferrari California Convertible 2012
439 BMW ActiveHybrid 5 Sedan 2012
440 Ford Edge SUV 2012
441 Audi S6 Sedan 2011
442 Cadillac SRX SUV 2012
443 Ferrari 458 Italia Coupe 2012
444 Audi S6 Sedan 2011
445 Ford F-150 Regular Cab 2007
446 Volkswagen Golf Hatchback 2012
447 Mercedes-Benz E-Class Sedan 2012
448 Bugatti Veyron 16.4 Coupe 2009
449 Audi A5 Coupe 2012
450 Audi TT Hatchback 2011
451 Bentley Continental GT Coupe 2007
452 Hyundai Tucson SUV 2012
453 Chevrolet HHR SS 2010
454 Chevrolet Silverado 1500 Hybrid Crew Cab 2012
455 Dodge Ram Pickup 3500 Crew Cab 2010
456 GMC Terra

850 Chevrolet Cobalt SS 2010
851 Chevrolet Monte Carlo Coupe 2007
852 Rolls-Royce Ghost Sedan 2012
853 Dodge Dakota Club Cab 2007
854 Jeep Liberty SUV 2012
855 Audi S5 Coupe 2012
856 BMW M5 Sedan 2010
857 Ford F-150 Regular Cab 2012
858 Ford Ranger SuperCab 2011
859 Chevrolet Sonic Sedan 2012
860 Nissan NV Passenger Van 2012
861 Nissan 240SX Coupe 1998
862 Chevrolet Silverado 1500 Classic Extended Cab 2007
863 Cadillac SRX SUV 2012
864 Ford F-450 Super Duty Crew Cab 2012
865 Ram C_V Cargo Van Minivan 2012
866 BMW M5 Sedan 2010
867 Suzuki SX4 Hatchback 2012
868 Audi 100 Wagon 1994
869 Audi R8 Coupe 2012
870 Volvo C30 Hatchback 2012
871 Nissan 240SX Coupe 1998
872 Chrysler Town and Country Minivan 2012
873 Jeep Compass SUV 2012
874 Toyota Sequoia SUV 2012
875 Dodge Durango SUV 2012
876 Hyundai Sonata Hybrid Sedan 2012
877 Ford GT Coupe 2006
878 Ford E-Series Wagon Van 2012
879 Lincoln Town Car Sedan 2011
880 Mercedes-Benz C-Class Sedan 2012
881 Volkswagen Beetle Hatchback 2012
882 Chevro

1245 Ford Edge SUV 2012
1246 smart fortwo Convertible 2012
1247 Dodge Dakota Crew Cab 2010
1248 Chevrolet Avalanche Crew Cab 2012
1249 Chevrolet HHR SS 2010
1250 Volkswagen Golf Hatchback 1991
1251 Chevrolet Tahoe Hybrid SUV 2012
1252 Dodge Durango SUV 2012
1253 Honda Accord Coupe 2012
1254 Infiniti QX56 SUV 2011
1255 Chevrolet Corvette Ron Fellows Edition Z06 2007
1256 Chrysler PT Cruiser Convertible 2008
1257 Plymouth Neon Coupe 1999
1258 Acura TL Type-S 2008
1259 Maybach Landaulet Convertible 2012
1260 Ford Mustang Convertible 2007
1261 BMW X5 SUV 2007
1262 Aston Martin Virage Coupe 2012
1263 BMW 6 Series Convertible 2007
1264 Rolls-Royce Phantom Drophead Coupe Convertible 2012
1265 Geo Metro Convertible 1993
1266 Infiniti QX56 SUV 2011
1267 Chrysler PT Cruiser Convertible 2008
1268 Acura TSX Sedan 2012
1269 Dodge Sprinter Cargo Van 2009
1270 BMW 6 Series Convertible 2007
1271 Jeep Liberty SUV 2012
1272 Toyota 4Runner SUV 2012
1273 Chevrolet Tahoe Hybrid SUV 2012
1274 Acura ZDX Hatc

1657 Ford F-150 Regular Cab 2012
1658 Buick Verano Sedan 2012
1659 Chrysler Sebring Convertible 2010
1660 Nissan Juke Hatchback 2012
1661 Chevrolet Malibu Sedan 2007
1662 Volkswagen Beetle Hatchback 2012
1663 Suzuki SX4 Hatchback 2012
1664 Chevrolet Monte Carlo Coupe 2007
1665 Hyundai Genesis Sedan 2012
1666 Jeep Wrangler SUV 2012
1667 BMW M3 Coupe 2012
1668 Mercedes-Benz C-Class Sedan 2012
1669 BMW 3 Series Wagon 2012
1670 Chevrolet Corvette Ron Fellows Edition Z06 2007
1671 Chevrolet Silverado 1500 Classic Extended Cab 2007
1672 Jeep Compass SUV 2012
1673 Dodge Challenger SRT8 2011
1674 Aston Martin Virage Convertible 2012
1675 Isuzu Ascender SUV 2008
1676 Audi 100 Wagon 1994
1677 Volkswagen Golf Hatchback 1991
1678 Chevrolet Express Van 2007
1679 Fisker Karma Sedan 2012
1680 Chevrolet Malibu Hybrid Sedan 2010
1681 Rolls-Royce Ghost Sedan 2012
1682 Suzuki Kizashi Sedan 2012
1683 Chevrolet Corvette Convertible 2012
1684 Toyota Camry Sedan 2012
1685 Hyundai Veracruz SUV 2012
1686 Dodge

2084 GMC Yukon Hybrid SUV 2012
2085 Ram C_V Cargo Van Minivan 2012
2086 Ford F-150 Regular Cab 2007
2087 Chevrolet Tahoe Hybrid SUV 2012
2088 Honda Odyssey Minivan 2012
2089 BMW X3 SUV 2012
2090 Hyundai Santa Fe SUV 2012
2091 Dodge Ram Pickup 3500 Quad Cab 2009
2092 Isuzu Ascender SUV 2008
2093 BMW X5 SUV 2007
2094 Lamborghini Diablo Coupe 2001
2095 Dodge Charger Sedan 2012
2096 Chevrolet Cobalt SS 2010
2097 Chevrolet Corvette Convertible 2012
2098 Infiniti G Coupe IPL 2012
2099 Audi A5 Coupe 2012
2100 Hyundai Santa Fe SUV 2012
2101 Aston Martin V8 Vantage Coupe 2012
2102 Dodge Magnum Wagon 2008
2103 Chevrolet Malibu Sedan 2007
2104 Spyker C8 Convertible 2009
2105 Jaguar XK XKR 2012
2106 Hyundai Veloster Hatchback 2012
2107 Audi A5 Coupe 2012
2108 Audi RS 4 Convertible 2008
2109 Dodge Charger Sedan 2012
2110 Maybach Landaulet Convertible 2012
2111 Aston Martin V8 Vantage Coupe 2012
2112 Bentley Continental GT Coupe 2007
2113 Ford E-Series Wagon Van 2012
2114 Dodge Magnum Wagon 2008
211

2503 Dodge Dakota Crew Cab 2010
2504 Ferrari FF Coupe 2012
2505 Bentley Continental GT Coupe 2012
2506 Chevrolet Corvette Ron Fellows Edition Z06 2007
2507 Isuzu Ascender SUV 2008
2508 Chevrolet Malibu Sedan 2007
2509 Cadillac SRX SUV 2012
2510 Chevrolet Malibu Hybrid Sedan 2010
2511 Jeep Liberty SUV 2012
2512 Suzuki Kizashi Sedan 2012
2513 Tesla Model S Sedan 2012
2514 Bentley Continental GT Coupe 2007
2515 HUMMER H3T Crew Cab 2010
2516 Mercedes-Benz E-Class Sedan 2012
2517 Acura TL Type-S 2008
2518 Chevrolet Camaro Convertible 2012
2519 Chevrolet Malibu Sedan 2007
2520 Acura TL Type-S 2008
2521 Ferrari California Convertible 2012
2522 Lamborghini Aventador Coupe 2012
2523 Ford GT Coupe 2006
2524 Volvo XC90 SUV 2007
2525 Hyundai Elantra Touring Hatchback 2012
2526 Ford Fiesta Sedan 2012
2527 Bentley Continental Supersports Conv. Convertible 2012
2528 Honda Odyssey Minivan 2007
2529 Isuzu Ascender SUV 2008
2530 Chevrolet Corvette ZR1 2012
2531 Suzuki SX4 Sedan 2012
2532 FIAT 500 Abarth

2923 Honda Odyssey Minivan 2012
2924 Ferrari 458 Italia Coupe 2012
2925 Ford Edge SUV 2012
2926 Dodge Charger SRT-8 2009
2927 Nissan 240SX Coupe 1998
2928 Fisker Karma Sedan 2012
2929 Spyker C8 Convertible 2009
2930 BMW M3 Coupe 2012
2931 Hyundai Azera Sedan 2012
2932 Chevrolet Avalanche Crew Cab 2012
2933 Bugatti Veyron 16.4 Coupe 2009
2934 Audi TT RS Coupe 2012
2935 Chevrolet TrailBlazer SS 2009
2936 FIAT 500 Convertible 2012
2937 Aston Martin Virage Coupe 2012
2938 Chevrolet Malibu Hybrid Sedan 2010
2939 Hyundai Genesis Sedan 2012
2940 Chevrolet Traverse SUV 2012
2941 Audi S6 Sedan 2011
2942 Chevrolet Silverado 1500 Regular Cab 2012
2943 Mercedes-Benz 300-Class Convertible 1993
2944 Acura ZDX Hatchback 2012
2945 Bentley Mulsanne Sedan 2011
2946 HUMMER H2 SUT Crew Cab 2009
2947 Hyundai Santa Fe SUV 2012
2948 Chevrolet Sonic Sedan 2012
2949 Bentley Continental Flying Spur Sedan 2007
2950 Ford Focus Sedan 2007
2951 Cadillac Escalade EXT Crew Cab 2007
2952 Tesla Model S Sedan 2012
2953 

3323 Jeep Grand Cherokee SUV 2012
3324 Chrysler 300 SRT-8 2010
3325 Acura RL Sedan 2012
3326 Volvo 240 Sedan 1993
3327 Dodge Caliber Wagon 2012
3328 Mercedes-Benz S-Class Sedan 2012
3329 Dodge Journey SUV 2012
3330 Audi S5 Coupe 2012
3331 Audi V8 Sedan 1994
3332 Mazda Tribute SUV 2011
3333 Chevrolet Impala Sedan 2007
3334 Chevrolet Silverado 1500 Regular Cab 2012
3335 Chevrolet Corvette Ron Fellows Edition Z06 2007
3336 Lincoln Town Car Sedan 2011
3337 Honda Odyssey Minivan 2012
3338 Ferrari 458 Italia Convertible 2012
3339 GMC Yukon Hybrid SUV 2012
3340 Toyota 4Runner SUV 2012
3341 Volkswagen Golf Hatchback 1991
3342 Ram C_V Cargo Van Minivan 2012
3343 GMC Terrain SUV 2012
3344 Ford E-Series Wagon Van 2012
3345 Mitsubishi Lancer Sedan 2012
3346 GMC Savana Van 2012
3347 Ford Ranger SuperCab 2011
3348 Hyundai Genesis Sedan 2012
3349 Chevrolet Camaro Convertible 2012
3350 Hyundai Genesis Sedan 2012
3351 Aston Martin V8 Vantage Coupe 2012
3352 Dodge Ram Pickup 3500 Crew Cab 2010
3353 Roll

3741 Toyota 4Runner SUV 2012
3742 smart fortwo Convertible 2012
3743 Ram C_V Cargo Van Minivan 2012
3744 Audi R8 Coupe 2012
3745 Chevrolet Express Van 2007
3746 Spyker C8 Convertible 2009
3747 Honda Accord Sedan 2012
3748 Acura Integra Type R 2001
3749 Ford Expedition EL SUV 2009
3750 Mercedes-Benz C-Class Sedan 2012
3751 BMW 1 Series Coupe 2012
3752 GMC Canyon Extended Cab 2012
3753 Aston Martin V8 Vantage Coupe 2012
3754 Eagle Talon Hatchback 1998
3755 Hyundai Elantra Touring Hatchback 2012
3756 BMW ActiveHybrid 5 Sedan 2012
3757 GMC Savana Van 2012
3758 Chevrolet Malibu Hybrid Sedan 2010
3759 Chevrolet Silverado 2500HD Regular Cab 2012
3760 Chevrolet Camaro Convertible 2012
3761 Hyundai Tucson SUV 2012
3762 Volkswagen Golf Hatchback 2012
3763 Bugatti Veyron 16.4 Coupe 2009
3764 Ford Ranger SuperCab 2011
3765 Isuzu Ascender SUV 2008
3766 Toyota Camry Sedan 2012
3767 Dodge Ram Pickup 3500 Crew Cab 2010
3768 HUMMER H3T Crew Cab 2010
3769 Dodge Caliber Wagon 2007
3770 Volvo C30 Hatchbac

4140 Mercedes-Benz Sprinter Van 2012
4141 Buick Rainier SUV 2007
4142 BMW 3 Series Wagon 2012
4143 Audi S4 Sedan 2007
4144 Bugatti Veyron 16.4 Convertible 2009
4145 Volvo XC90 SUV 2007
4146 GMC Savana Van 2012
4147 Scion xD Hatchback 2012
4148 BMW 1 Series Convertible 2012
4149 Geo Metro Convertible 1993
4150 Cadillac Escalade EXT Crew Cab 2007
4151 Hyundai Genesis Sedan 2012
4152 Ferrari 458 Italia Coupe 2012
4153 Lincoln Town Car Sedan 2011
4154 HUMMER H3T Crew Cab 2010
4155 GMC Acadia SUV 2012
4156 Mercedes-Benz 300-Class Convertible 1993
4157 Ford Freestar Minivan 2007
4158 Spyker C8 Coupe 2009
4159 Cadillac SRX SUV 2012
4160 Dodge Caliber Wagon 2007
4161 Chevrolet Corvette Convertible 2012
4162 Ford GT Coupe 2006
4163 Bentley Arnage Sedan 2009
4164 Dodge Journey SUV 2012
4165 Audi S5 Convertible 2012
4166 Jeep Grand Cherokee SUV 2012
4167 Dodge Ram Pickup 3500 Crew Cab 2010
4168 Audi TT Hatchback 2011
4169 Audi TT Hatchback 2011
4170 Chevrolet Traverse SUV 2012
4171 Dodge Magnum W

4536 BMW 1 Series Convertible 2012
4537 Ford F-150 Regular Cab 2007
4538 Suzuki Aerio Sedan 2007
4539 Aston Martin V8 Vantage Coupe 2012
4540 BMW X3 SUV 2012
4541 Mercedes-Benz Sprinter Van 2012
4542 Mercedes-Benz S-Class Sedan 2012
4543 Jaguar XK XKR 2012
4544 Infiniti QX56 SUV 2011
4545 Audi S5 Coupe 2012
4546 Audi 100 Wagon 1994
4547 Toyota Corolla Sedan 2012
4548 Chrysler 300 SRT-8 2010
4549 Chrysler 300 SRT-8 2010
4550 Chevrolet Silverado 1500 Hybrid Crew Cab 2012
4551 Toyota Camry Sedan 2012
4552 Cadillac SRX SUV 2012
4553 Toyota Sequoia SUV 2012
4554 Chevrolet Tahoe Hybrid SUV 2012
4555 BMW 1 Series Coupe 2012
4556 Mazda Tribute SUV 2011
4557 BMW 6 Series Convertible 2007
4558 Aston Martin Virage Coupe 2012
4559 Dodge Durango SUV 2012
4560 Lamborghini Diablo Coupe 2001
4561 Audi 100 Sedan 1994
4562 Audi V8 Sedan 1994
4563 Spyker C8 Convertible 2009
4564 Nissan NV Passenger Van 2012
4565 Chevrolet HHR SS 2010
4566 Toyota Sequoia SUV 2012
4567 Chevrolet Silverado 1500 Extended Cab

4892 Aston Martin V8 Vantage Coupe 2012
4893 Audi S5 Coupe 2012
4894 Spyker C8 Convertible 2009
4895 Hyundai Genesis Sedan 2012
4896 Bentley Mulsanne Sedan 2011
4897 Chrysler 300 SRT-8 2010
4898 Chevrolet Traverse SUV 2012
4899 McLaren MP4-12C Coupe 2012
4900 Chevrolet Express Van 2007
4901 Jeep Wrangler SUV 2012
4902 Volvo 240 Sedan 1993
4903 Jeep Liberty SUV 2012
4904 Aston Martin V8 Vantage Convertible 2012
4905 Toyota Corolla Sedan 2012
4906 Cadillac SRX SUV 2012
4907 Volkswagen Golf Hatchback 2012
4908 Dodge Caliber Wagon 2012
4909 Suzuki SX4 Hatchback 2012
4910 Ford F-150 Regular Cab 2007
4911 Chevrolet Corvette ZR1 2012
4912 BMW ActiveHybrid 5 Sedan 2012
4913 McLaren MP4-12C Coupe 2012
4914 Volvo 240 Sedan 1993
4915 BMW M5 Sedan 2010
4916 Nissan Leaf Hatchback 2012
4917 Ferrari 458 Italia Convertible 2012
4918 Dodge Durango SUV 2007
4919 Lincoln Town Car Sedan 2011
4920 Acura RL Sedan 2012
4921 Rolls-Royce Phantom Drophead Coupe Convertible 2012
4922 Buick Regal GS 2012
4923 Aud

5294 Bentley Continental GT Coupe 2012
5295 Honda Accord Sedan 2012
5296 Hyundai Veracruz SUV 2012
5297 Chevrolet Malibu Sedan 2007
5298 Aston Martin V8 Vantage Convertible 2012
5299 Audi TT Hatchback 2011
5300 Chevrolet Monte Carlo Coupe 2007
5301 Mercedes-Benz Sprinter Van 2012
5302 Suzuki SX4 Hatchback 2012
5303 Porsche Panamera Sedan 2012
5304 Suzuki Aerio Sedan 2007
5305 Suzuki SX4 Hatchback 2012
5306 Mercedes-Benz S-Class Sedan 2012
5307 Toyota Sequoia SUV 2012
5308 Mercedes-Benz Sprinter Van 2012
5309 Lamborghini Aventador Coupe 2012
5310 GMC Yukon Hybrid SUV 2012
5311 BMW 3 Series Sedan 2012
5312 Buick Regal GS 2012
5313 Audi A5 Coupe 2012
5314 Ford Mustang Convertible 2007
5315 Audi TTS Coupe 2012
5316 Ferrari 458 Italia Convertible 2012
5317 Mercedes-Benz 300-Class Convertible 1993
5318 Chrysler 300 SRT-8 2010
5319 Audi 100 Sedan 1994
5320 Maybach Landaulet Convertible 2012
5321 Jeep Patriot SUV 2012
5322 Dodge Caliber Wagon 2007
5323 Chrysler PT Cruiser Convertible 2008
5324

5692 Chevrolet Camaro Convertible 2012
5693 Dodge Durango SUV 2012
5694 Volkswagen Beetle Hatchback 2012
5695 Suzuki SX4 Hatchback 2012
5696 GMC Terrain SUV 2012
5697 Buick Regal GS 2012
5698 Chevrolet Tahoe Hybrid SUV 2012
5699 BMW M3 Coupe 2012
5700 Dodge Dakota Club Cab 2007
5701 Bentley Mulsanne Sedan 2011
5702 Isuzu Ascender SUV 2008
5703 Hyundai Sonata Hybrid Sedan 2012
5704 Cadillac CTS-V Sedan 2012
5705 Lincoln Town Car Sedan 2011
5706 smart fortwo Convertible 2012
5707 Ford Expedition EL SUV 2009
5708 Audi 100 Wagon 1994
5709 Chevrolet Sonic Sedan 2012
5710 Ford E-Series Wagon Van 2012
5711 Chevrolet Sonic Sedan 2012
5712 Chevrolet Silverado 1500 Regular Cab 2012
5713 Ford Expedition EL SUV 2009
5714 Ford Mustang Convertible 2007
5715 Ford Mustang Convertible 2007
5716 Ford Expedition EL SUV 2009
5717 Hyundai Sonata Hybrid Sedan 2012
5718 Ram C_V Cargo Van Minivan 2012
5719 Audi 100 Wagon 1994
5720 Dodge Dakota Club Cab 2007
5721 Audi RS 4 Convertible 2008
5722 Rolls-Royce Pha

6099 BMW Z4 Convertible 2012
6100 BMW M3 Coupe 2012
6101 Rolls-Royce Phantom Drophead Coupe Convertible 2012
6102 Land Rover LR2 SUV 2012
6103 Rolls-Royce Phantom Sedan 2012
6104 Ford Freestar Minivan 2007
6105 Chevrolet Silverado 1500 Extended Cab 2012
6106 Dodge Dakota Club Cab 2007
6107 Ford GT Coupe 2006
6108 Ferrari 458 Italia Coupe 2012
6109 Chevrolet Corvette Convertible 2012
6110 Volvo XC90 SUV 2007
6111 Hyundai Veracruz SUV 2012
6112 Audi TT Hatchback 2011
6113 Dodge Magnum Wagon 2008
6114 Audi S5 Coupe 2012
6115 Hyundai Azera Sedan 2012
6116 Dodge Durango SUV 2007
6117 Volkswagen Beetle Hatchback 2012
6118 Volkswagen Beetle Hatchback 2012
6119 Acura RL Sedan 2012
6120 Cadillac Escalade EXT Crew Cab 2007
6121 GMC Yukon Hybrid SUV 2012
6122 BMW X6 SUV 2012
6123 Cadillac Escalade EXT Crew Cab 2007
6124 Volkswagen Beetle Hatchback 2012
6125 Chrysler Aspen SUV 2009
6126 Daewoo Nubira Wagon 2002
6127 Dodge Ram Pickup 3500 Crew Cab 2010
6128 Hyundai Elantra Touring Hatchback 2012
61

6496 Bentley Continental Flying Spur Sedan 2007
6497 Chrysler Aspen SUV 2009
6498 Hyundai Tucson SUV 2012
6499 Mercedes-Benz E-Class Sedan 2012
6500 Chevrolet HHR SS 2010
6501 Bugatti Veyron 16.4 Coupe 2009
6502 Ford Edge SUV 2012
6503 Honda Accord Coupe 2012
6504 Jeep Compass SUV 2012
6505 GMC Savana Van 2012
6506 BMW X6 SUV 2012
6507 Jeep Wrangler SUV 2012
6508 Isuzu Ascender SUV 2008
6509 Land Rover Range Rover SUV 2012
6510 Chevrolet TrailBlazer SS 2009
6511 Ford Focus Sedan 2007
6512 Chevrolet Silverado 1500 Hybrid Crew Cab 2012
6513 Chevrolet Silverado 1500 Hybrid Crew Cab 2012
6514 Audi S5 Coupe 2012
6515 Rolls-Royce Phantom Sedan 2012
6516 Dodge Challenger SRT8 2011
6517 BMW X3 SUV 2012
6518 Ferrari FF Coupe 2012
6519 Bentley Arnage Sedan 2009
6520 Dodge Challenger SRT8 2011
6521 Bentley Continental Flying Spur Sedan 2007
6522 Lincoln Town Car Sedan 2011
6523 Audi TT RS Coupe 2012
6524 Toyota 4Runner SUV 2012
6525 Chevrolet Express Van 2007
6526 Bentley Mulsanne Sedan 2011
6527

6882 BMW M6 Convertible 2010
6883 Eagle Talon Hatchback 1998
6884 BMW X5 SUV 2007
6885 Hyundai Veloster Hatchback 2012
6886 smart fortwo Convertible 2012
6887 Rolls-Royce Ghost Sedan 2012
6888 Chevrolet Malibu Hybrid Sedan 2010
6889 Jeep Patriot SUV 2012
6890 Mercedes-Benz E-Class Sedan 2012
6891 Ferrari FF Coupe 2012
6892 Volkswagen Golf Hatchback 2012
6893 Spyker C8 Convertible 2009
6894 Hyundai Elantra Touring Hatchback 2012
6895 Chevrolet Silverado 1500 Regular Cab 2012
6896 Jeep Compass SUV 2012
6897 BMW M6 Convertible 2010
6898 Jeep Liberty SUV 2012
6899 Ferrari FF Coupe 2012
6900 Chrysler Crossfire Convertible 2008
6901 Chevrolet HHR SS 2010
6902 Buick Regal GS 2012
6903 BMW 3 Series Wagon 2012
6904 Isuzu Ascender SUV 2008
6905 Audi TTS Coupe 2012
6906 Honda Odyssey Minivan 2012
6907 Hyundai Sonata Hybrid Sedan 2012
6908 Ford Freestar Minivan 2007
6909 Tesla Model S Sedan 2012
6910 Volkswagen Beetle Hatchback 2012
6911 Audi V8 Sedan 1994
6912 Audi V8 Sedan 1994
6913 BMW 6 Series

7291 BMW 3 Series Sedan 2012
7292 GMC Terrain SUV 2012
7293 Chrysler Aspen SUV 2009
7294 Jeep Wrangler SUV 2012
7295 Chrysler 300 SRT-8 2010
7296 Ford E-Series Wagon Van 2012
7297 Buick Enclave SUV 2012
7298 Ford E-Series Wagon Van 2012
7299 Land Rover LR2 SUV 2012
7300 Mercedes-Benz S-Class Sedan 2012
7301 Audi S6 Sedan 2011
7302 Chevrolet Silverado 1500 Regular Cab 2012
7303 Jeep Compass SUV 2012
7304 Ford F-450 Super Duty Crew Cab 2012
7305 Tesla Model S Sedan 2012
7306 Chrysler 300 SRT-8 2010
7307 Aston Martin V8 Vantage Convertible 2012
7308 GMC Savana Van 2012
7309 Dodge Caliber Wagon 2012
7310 Audi S5 Convertible 2012
7311 Nissan Juke Hatchback 2012
7312 Suzuki Aerio Sedan 2007
7313 Daewoo Nubira Wagon 2002
7314 Chevrolet Silverado 1500 Hybrid Crew Cab 2012
7315 HUMMER H3T Crew Cab 2010
7316 Hyundai Genesis Sedan 2012
7317 Suzuki Aerio Sedan 2007
7318 Bugatti Veyron 16.4 Coupe 2009
7319 Lamborghini Diablo Coupe 2001
7320 Ferrari California Convertible 2012
7321 Dodge Dakota Club

7672 Dodge Ram Pickup 3500 Quad Cab 2009
7673 Ford Freestar Minivan 2007
7674 BMW Z4 Convertible 2012
7675 Dodge Caravan Minivan 1997
7676 Rolls-Royce Phantom Sedan 2012
7677 Audi RS 4 Convertible 2008
7678 Daewoo Nubira Wagon 2002
7679 Hyundai Tucson SUV 2012
7680 Cadillac SRX SUV 2012
7681 Audi V8 Sedan 1994
7682 Dodge Challenger SRT8 2011
7683 Chevrolet Corvette ZR1 2012
7684 Dodge Durango SUV 2007
7685 Chevrolet Avalanche Crew Cab 2012
7686 Hyundai Veloster Hatchback 2012
7687 Jeep Grand Cherokee SUV 2012
7688 Dodge Caliber Wagon 2007
7689 Lamborghini Diablo Coupe 2001
7690 Volvo 240 Sedan 1993
7691 Volkswagen Beetle Hatchback 2012
7692 Ford E-Series Wagon Van 2012
7693 Audi A5 Coupe 2012
7694 Ford Expedition EL SUV 2009
7695 Hyundai Santa Fe SUV 2012
7696 Hyundai Tucson SUV 2012
7697 Suzuki SX4 Hatchback 2012
7698 GMC Yukon Hybrid SUV 2012
7699 Chevrolet Express Cargo Van 2007
7700 Audi V8 Sedan 1994
7701 Toyota 4Runner SUV 2012
7702 Chevrolet Silverado 1500 Regular Cab 2012
7703 

#### 2.1 Move Test to Train (add more data to training set)

In [4]:
import math
import os
path_train = 'data/train/'
path_valid = 'data/valid/'
folders = os.listdir(path_valid)
for folder_name in folders:
    if os.path.isdir(path_valid+folder_name):
        full_folder_name = path_valid+folder_name
        
        num_images = len([name for name in os.listdir(full_folder_name) if os.path.isfile(os.path.join(full_folder_name, name))])
        num_val = math.floor(num_images/2.5)
        print(num_val)
        for i, file in enumerate(os.listdir(full_folder_name)):
            if i < num_val:
                file_valid = path_valid+folder_name+'/'+file
                file_train = path_train+folder_name+'/'+file
                print(file_valid, file_train)
                shutil.move(file_valid, file_train)
                

12
data/valid/Infiniti QX56 SUV 2011/01956.jpg data/train/Infiniti QX56 SUV 2011/01956.jpg
data/valid/Infiniti QX56 SUV 2011/00478.jpg data/train/Infiniti QX56 SUV 2011/00478.jpg
data/valid/Infiniti QX56 SUV 2011/03223.jpg data/train/Infiniti QX56 SUV 2011/03223.jpg
data/valid/Infiniti QX56 SUV 2011/07080.jpg data/train/Infiniti QX56 SUV 2011/07080.jpg
data/valid/Infiniti QX56 SUV 2011/04324.jpg data/train/Infiniti QX56 SUV 2011/04324.jpg
data/valid/Infiniti QX56 SUV 2011/07934.jpg data/train/Infiniti QX56 SUV 2011/07934.jpg
data/valid/Infiniti QX56 SUV 2011/00491.jpg data/train/Infiniti QX56 SUV 2011/00491.jpg
data/valid/Infiniti QX56 SUV 2011/02499.jpg data/train/Infiniti QX56 SUV 2011/02499.jpg
data/valid/Infiniti QX56 SUV 2011/04803.jpg data/train/Infiniti QX56 SUV 2011/04803.jpg
data/valid/Infiniti QX56 SUV 2011/01641.jpg data/train/Infiniti QX56 SUV 2011/01641.jpg
data/valid/Infiniti QX56 SUV 2011/01068.jpg data/train/Infiniti QX56 SUV 2011/01068.jpg
data/valid/Infiniti QX56 SUV 

data/valid/Audi 100 Sedan 1994/03791.jpg data/train/Audi 100 Sedan 1994/03791.jpg
data/valid/Audi 100 Sedan 1994/05077.jpg data/train/Audi 100 Sedan 1994/05077.jpg
data/valid/Audi 100 Sedan 1994/00333.jpg data/train/Audi 100 Sedan 1994/00333.jpg
data/valid/Audi 100 Sedan 1994/04561.jpg data/train/Audi 100 Sedan 1994/04561.jpg
data/valid/Audi 100 Sedan 1994/07284.jpg data/train/Audi 100 Sedan 1994/07284.jpg
data/valid/Audi 100 Sedan 1994/02187.jpg data/train/Audi 100 Sedan 1994/02187.jpg
data/valid/Audi 100 Sedan 1994/04250.jpg data/train/Audi 100 Sedan 1994/04250.jpg
data/valid/Audi 100 Sedan 1994/07491.jpg data/train/Audi 100 Sedan 1994/07491.jpg
data/valid/Audi 100 Sedan 1994/00573.jpg data/train/Audi 100 Sedan 1994/00573.jpg
data/valid/Audi 100 Sedan 1994/06767.jpg data/train/Audi 100 Sedan 1994/06767.jpg
data/valid/Audi 100 Sedan 1994/00082.jpg data/train/Audi 100 Sedan 1994/00082.jpg
data/valid/Audi 100 Sedan 1994/06963.jpg data/train/Audi 100 Sedan 1994/06963.jpg
data/valid/Audi 

data/valid/Lamborghini Diablo Coupe 2001/07480.jpg data/train/Lamborghini Diablo Coupe 2001/07480.jpg
data/valid/Lamborghini Diablo Coupe 2001/06707.jpg data/train/Lamborghini Diablo Coupe 2001/06707.jpg
data/valid/Lamborghini Diablo Coupe 2001/02094.jpg data/train/Lamborghini Diablo Coupe 2001/02094.jpg
data/valid/Lamborghini Diablo Coupe 2001/01927.jpg data/train/Lamborghini Diablo Coupe 2001/01927.jpg
data/valid/Lamborghini Diablo Coupe 2001/03072.jpg data/train/Lamborghini Diablo Coupe 2001/03072.jpg
18
data/valid/Volkswagen Golf Hatchback 1991/06330.jpg data/train/Volkswagen Golf Hatchback 1991/06330.jpg
data/valid/Volkswagen Golf Hatchback 1991/00053.jpg data/train/Volkswagen Golf Hatchback 1991/00053.jpg
data/valid/Volkswagen Golf Hatchback 1991/01341.jpg data/train/Volkswagen Golf Hatchback 1991/01341.jpg
data/valid/Volkswagen Golf Hatchback 1991/01377.jpg data/train/Volkswagen Golf Hatchback 1991/01377.jpg
data/valid/Volkswagen Golf Hatchback 1991/00943.jpg data/train/Volkswag

data/valid/Nissan Leaf Hatchback 2012/04634.jpg data/train/Nissan Leaf Hatchback 2012/04634.jpg
data/valid/Nissan Leaf Hatchback 2012/03559.jpg data/train/Nissan Leaf Hatchback 2012/03559.jpg
data/valid/Nissan Leaf Hatchback 2012/06166.jpg data/train/Nissan Leaf Hatchback 2012/06166.jpg
data/valid/Nissan Leaf Hatchback 2012/00050.jpg data/train/Nissan Leaf Hatchback 2012/00050.jpg
data/valid/Nissan Leaf Hatchback 2012/07942.jpg data/train/Nissan Leaf Hatchback 2012/07942.jpg
data/valid/Nissan Leaf Hatchback 2012/04940.jpg data/train/Nissan Leaf Hatchback 2012/04940.jpg
data/valid/Nissan Leaf Hatchback 2012/03517.jpg data/train/Nissan Leaf Hatchback 2012/03517.jpg
data/valid/Nissan Leaf Hatchback 2012/07270.jpg data/train/Nissan Leaf Hatchback 2012/07270.jpg
data/valid/Nissan Leaf Hatchback 2012/08004.jpg data/train/Nissan Leaf Hatchback 2012/08004.jpg
data/valid/Nissan Leaf Hatchback 2012/07221.jpg data/train/Nissan Leaf Hatchback 2012/07221.jpg
data/valid/Nissan Leaf Hatchback 2012/03

data/valid/Acura ZDX Hatchback 2012/03059.jpg data/train/Acura ZDX Hatchback 2012/03059.jpg
17
data/valid/Bugatti Veyron 16.4 Coupe 2009/04787.jpg data/train/Bugatti Veyron 16.4 Coupe 2009/04787.jpg
data/valid/Bugatti Veyron 16.4 Coupe 2009/00448.jpg data/train/Bugatti Veyron 16.4 Coupe 2009/00448.jpg
data/valid/Bugatti Veyron 16.4 Coupe 2009/01788.jpg data/train/Bugatti Veyron 16.4 Coupe 2009/01788.jpg
data/valid/Bugatti Veyron 16.4 Coupe 2009/03763.jpg data/train/Bugatti Veyron 16.4 Coupe 2009/03763.jpg
data/valid/Bugatti Veyron 16.4 Coupe 2009/02681.jpg data/train/Bugatti Veyron 16.4 Coupe 2009/02681.jpg
data/valid/Bugatti Veyron 16.4 Coupe 2009/05134.jpg data/train/Bugatti Veyron 16.4 Coupe 2009/05134.jpg
data/valid/Bugatti Veyron 16.4 Coupe 2009/02392.jpg data/train/Bugatti Veyron 16.4 Coupe 2009/02392.jpg
data/valid/Bugatti Veyron 16.4 Coupe 2009/03619.jpg data/train/Bugatti Veyron 16.4 Coupe 2009/03619.jpg
data/valid/Bugatti Veyron 16.4 Coupe 2009/03660.jpg data/train/Bugatti Ve

### Convertible
Aston Martin V8 Vantage Convertible 2012
Aston Martin Virage Convertible 2012
Audi RS 4 Convertible 2008
Audi S5 Convertible 2012
Bentley Continental Supersports Conv. Convertible 2012
BMW 1 Series Convertible 2012
BMW 6 Series Convertible 2007
BMW M6 Convertible 2010
BMW Z4 Convertible 2012
Bugatti Veyron 16.4 Convertible 2009
Chevrolet Camaro Convertible 2012
Chevrolet Corvette Convertible 2012
Chrysler Crossfire Convertible 2008
Chrysler PT Cruiser Convertible 2008
Chrysler Sebring Convertible 2010
Ferrari 458 Italia Convertible 2012
Ferrari California Convertible 2012
FIAT 500 Convertible 2012
Ford Mustang Convertible 2007
Geo Metro Convertible 1993
Jaguar XK XKR 2012
Maybach Landaulet Convertible 2012
Mercedes-Benz 300-Class Convertible 1993
MINI Cooper Roadster Convertible 2012
Rolls-Royce Phantom Drophead Coupe Convertible 2012
smart fortwo Convertible 2012
Spyker C8 Convertible 2009

### Coupe
Acura Integra Type R 2001
Aston Martin V8 Vantage Coupe 2012
Aston Martin Virage Coupe 2012
Audi A5 Coupe 2012
Audi R8 Coupe 2012
Audi S5 Coupe 2012
Audi TT RS Coupe 2012
Audi TTS Coupe 2012
Bentley Continental GT Coupe 2007
Bentley Continental GT Coupe 2012
BMW 1 Series Coupe 2012
BMW M3 Coupe 2012
Bugatti Veyron 16.4 Coupe 2009
Chevrolet Cobalt SS 2010
Chevrolet Corvette Ron Fellows Edition Z06 2007
Chevrolet Corvette ZR1 2012
Chevrolet Monte Carlo Coupe 2007
Dodge Challenger SRT8 2011
Ferrari 458 Italia Coupe 2012
Ferrari FF Coupe 2012
Ford GT Coupe 2006
Honda Accord Coupe 2012
Infiniti G Coupe IPL 2012
Lamborghini Aventador Coupe 2012
Lamborghini Diablo Coupe 2001
Lamborghini Gallardo LP 570-4 Superleggera 2012
Lamborghini Reventon Coupe 2008
McLaren MP4-12C Coupe 2012
Mercedes-Benz SL-Class Coupe 2009
Nissan 240SX Coupe 1998
Plymouth Neon Coupe 1999
Spyker C8 Coupe 2009

### Sedan
Acura RL Sedan 2012
Acura TL Sedan 2012
Acura TL Type-S 2008
Acura TSX Sedan 2012
Audi 100 Sedan 1994
Audi S4 Sedan 2007
Audi S4 Sedan 2012
Audi S6 Sedan 2011
Audi V8 Sedan 1994
Bentley Arnage Sedan 2009
Bentley Continental Flying Spur Sedan 2007
Bentley Mulsanne Sedan 2011
BMW 3 Series Sedan 2012
BMW ActiveHybrid 5 Sedan 2012
BMW M5 Sedan 2010
Buick Regal GS 2012
Buick Verano Sedan 2012
Cadillac CTS-V Sedan 2012
Chevrolet Impala Sedan 2007
Chevrolet Malibu Hybrid Sedan 2010
Chevrolet Malibu Sedan 2007
Chevrolet Sonic Sedan 2012
Chrysler 300 SRT-8 2010
Dodge Charger Sedan 2012
Dodge Charger SRT-8 2009
Fisker Karma Sedan 2012
Ford Fiesta Sedan 2012
Ford Focus Sedan 2007
Honda Accord Sedan 2012
Hyundai Accent Sedan 2012
Hyundai Azera Sedan 2012
Hyundai Elantra Sedan 2007
Hyundai Genesis Sedan 2012
Hyundai Sonata Hybrid Sedan 2012
Hyundai Sonata Sedan 2012
Lincoln Town Car Sedan 2011
Mercedes-Benz C-Class Sedan 2012
Mercedes-Benz E-Class Sedan 2012
Mercedes-Benz S-Class Sedan 2012
Mitsubishi Lancer Sedan 2012
Porsche Panamera Sedan 2012
Rolls-Royce Ghost Sedan 2012
Rolls-Royce Phantom Sedan 2012
Suzuki Aerio Sedan 2007
Suzuki Kizashi Sedan 2012
Suzuki SX4 Sedan 2012
Tesla Model S Sedan 2012
Toyota Camry Sedan 2012
Toyota Corolla Sedan 2012
Volvo 240 Sedan 1993

### Hatchback
Acura ZDX Hatchback 2012
Audi TT Hatchback 2011
Eagle Talon Hatchback 1998
Hyundai Elantra Touring Hatchback 2012
Hyundai Veloster Hatchback 2012
Nissan Juke Hatchback 2012
Nissan Leaf Hatchback 2012
Scion xD Hatchback 2012
Suzuki SX4 Hatchback 2012
Volkswagen Beetle Hatchback 2012
Volkswagen Golf Hatchback 1991
Volkswagen Golf Hatchback 2012
Volvo C30 Hatchback 2012

### SUV/Wagon
AM General Hummer SUV 2000
Audi 100 Wagon 1994
BMW 3 Series Wagon 2012
BMW X3 SUV 2012
BMW X5 SUV 2007
BMW X6 SUV 2012
Buick Enclave SUV 2012
Buick Rainier SUV 2007
Cadillac SRX SUV 2012
Chevrolet Tahoe Hybrid SUV 2012
Chevrolet TrailBlazer SS 2009
Chevrolet Traverse SUV 2012
Chrysler Aspen SUV 2009
Daewoo Nubira Wagon 2002
Dodge Caliber Wagon 2007
Dodge Caliber Wagon 2012
Dodge Durango SUV 2007
Dodge Durango SUV 2012
Dodge Journey SUV 2012
Dodge Magnum Wagon 2008
Ford Edge SUV 2012
Ford Expedition EL SUV 2009
GMC Acadia SUV 2012
GMC Terrain SUV 2012
GMC Yukon Hybrid SUV 2012
Hyundai Santa Fe SUV 2012
Hyundai Tucson SUV 2012
Hyundai Veracruz SUV 2012
Infiniti QX56 SUV 2011
Isuzu Ascender SUV 2008
Jeep Compass SUV 2012
Jeep Grand Cherokee SUV 2012
Jeep Liberty SUV 2012
Jeep Patriot SUV 2012
Jeep Wrangler SUV 2012
Land Rover LR2 SUV 2012
Land Rover Range Rover SUV 2012
Mazda Tribute SUV 2011
Toyota 4Runner SUV 2012
Toyota Sequoia SUV 2012
Volvo XC90 SUV 2007

### Truck/Cab
Cadillac Escalade EXT Crew Cab 2007
Chevrolet Avalanche Crew Cab 2012
Chevrolet Silverado 1500 Classic Extended Cab 2007
Chevrolet Silverado 1500 Extended Cab 2012
Chevrolet Silverado 1500 Hybrid Crew Cab 2012
Chevrolet Silverado 1500 Regular Cab 2012
Chevrolet Silverado 2500HD Regular Cab 2012
Dodge Dakota Club Cab 2007
Dodge Dakota Crew Cab 2010
Dodge Ram Pickup 3500 Crew Cab 2010
Dodge Ram Pickup 3500 Quad Cab 2009
Ford F-150 Regular Cab 2007
Ford F-150 Regular Cab 2012
Ford F-450 Super Duty Crew Cab 2012
Ford Ranger SuperCab 2011
GMC Canyon Extended Cab 2012
HUMMER H2 SUT Crew Cab 2009
HUMMER H3T Crew Cab 2010

### Van
Chevrolet Express Cargo Van 2007
Chevrolet Express Van 2007
Chevrolet HHR SS 2010
Chrysler Town and Country Minivan 2012
Dodge Caravan Minivan 1997
Dodge Sprinter Cargo Van 2009
Ford E-Series Wagon Van 2012
Ford Freestar Minivan 2007
GMC Savana Van 2012
Honda Odyssey Minivan 2007
Honda Odyssey Minivan 2012
Mercedes-Benz Sprinter Van 2012
Nissan NV Passenger Van 2012
Ram C_V Cargo Van Minivan 2012

### Other
FIAT 500 Abarth 2012

### 3. Merge to Each Type (eg. sedan, convertable, van...)

In [5]:
file = open('class_data.rtf', 'r')
text = file.read().strip()
file.close()
text = text.replace('{\\rtf1\\ansi\\ansicpg1252\\cocoartf1504\\cocoasubrtf830\n{\\fonttbl\\f0\\fswiss\\fcharset0 Helvetica;}\n{\\colortbl;\\red255\\green255\\blue255;}\n{\\*\\expandedcolortbl;;}\n\\margl1440\\margr1440\\vieww10800\\viewh8400\\viewkind0\n\\pard\\tx720\\tx1440\\tx2160\\tx2880\\tx3600\\tx4320\\tx5040\\tx5760\\tx6480\\tx7200\\tx7920\\tx8640\\pardirnatural\\partightenfactor0\n\n\\f0\\fs24 \\cf0 ','')
text = text[:-1]
text = text.replace('\\\n', '\n')
text_array = text.split('\n\n')

def move_to_type_by_data_set(data_folder):
    temp_folder = 'data/temp/'
    for group_text in text_array:
        vehicle_types = group_text.split('\n')
        head = vehicle_types.pop(0).replace('###', '')

        if head != 'other': # other type is very confusing so exclude it
            pathlib.Path(temp_folder+head).mkdir(parents=True, exist_ok=True)
            for folder in vehicle_types:
                for file in os.listdir(data_folder+folder):
                    shutil.move(data_folder+folder+'/'+file, temp_folder+head+'/'+folder+file)
    shutil.rmtree(data_folder)
    shutil.move(temp_folder, data_folder)

move_to_type_by_data_set('data/train/')
move_to_type_by_data_set('data/valid/')

### Show All Number of Data

In [6]:
def get_num_data(path):
    return sum([len(os.listdir(path+folder)) for folder in os.listdir(path)])
print(get_num_data('data/train/'))
print(get_num_data('data/valid/'))

11218
4890
